File: backend/app/schemas/subscription.py
from datetime import date
from uuid import UUID
from pydantic import BaseModel, Field


class PlanResponse(BaseModel):
    id: UUID
    name: str
    slug: str
    plan_type: str
    duration_type: str
    duration_days: int
    price: float
    currency: str
    features: str | None
    max_job_postings_per_month: int
    max_applications_per_month: int
    can_view_contact_info: bool
    can_see_premium_jobs: bool
    priority_listing: bool
    sort_order: int

    class Config:
        from_attributes = True


class OfferResponse(BaseModel):
    id: UUID
    offer_name: str
    offer_code: str
    discount_type: str
    discount_value: float
    valid_until: date | None

    class Config:
        from_attributes = True


class PlanWithOffersResponse(PlanResponse):
    offers: list[OfferResponse] = []


class SubscribeRequest(BaseModel):
    plan_id: UUID
    offer_code: str | None = None


class SubscribeResponse(BaseModel):
    order_id: str
    amount: float
    currency: str
    razorpay_order_id: str


class VerifyPaymentRequest(BaseModel):
    razorpay_order_id: str
    razorpay_payment_id: str
    razorpay_signature: str


class MySubscriptionResponse(BaseModel):
    plan_name: str
    status: str
    start_date: date
    end_date: date
    auto_renew: bool
    amount_paid: float | None
    currency: str | None

File: saralam_webapp/src/types/subscription.ts
export interface SubscriptionPlan {
  id: string
  name: string
  slug: string
  plan_type: string
  duration_type: string
  duration_days: number
  price: number
  currency: string
  features?: string
  max_job_postings_per_month: number
  max_applications_per_month: number
  can_view_contact_info: boolean
  can_see_premium_jobs: boolean
  priority_listing: boolean
  sort_order: number
  offers?: SubscriptionOffer[]
}

export interface SubscriptionOffer {
  id: string
  offer_name: string
  offer_code: string
  discount_type: string
  discount_value: number
  valid_until?: string
}

export interface SubscriptionMy {
  plan_id: string
  plan_name: string
  status: string
  current_period_end?: string
  jobs_posted_this_month?: number
  jobs_limit_per_month?: number
  applications_submitted_this_month?: number
  applications_limit_per_month?: number
}

export interface PaymentHistoryItem {
  id: string
  date: string
  plan_name: string
  amount: number
  currency: string
  status: string
  receipt_url?: string
}

export interface RazorpayOrder {
  razorpay_order_id: string
  amount: number
  currency: string
  plan_name: string
  plan_id: string
}
